### PR TITLE
Revert "Revert MPS changes (#78335)"

### PR DESCRIPTION
**Pull request by kulinseth**
This reverts commit ffb31014847faccb724ef70924ba936b75144c85.

Add back the changes. Fix the build break.
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by danieldk** [comment]
There seems to be a syntax error in these changes. I had to remove a parenthesis:

```
diff --git a/aten/src/ATen/mps/MPSDevice.mm b/aten/src/ATen/mps/MPSDevice.mm
index ecf0755a10..aea575f854 100644
--- a/aten/src/ATen/mps/MPSDevice.mm
+++ b/aten/src/ATen/mps/MPSDevice.mm
@@ -32,7 +32,7 @@
                                                                  initState:
                                                                   initCell:
                                                                 descriptor:
-                                                           name:)] == NO)) {
+                                                           name:)] == NO) {
     return;
   }
   NSArray* devices = [MTLCopyAllDevices() autorelease];
```

File: aten/src/ATen/mps/MPSDevice.mm [hunk 1]
```diff
@@ -22,10 +22,17 @@
...
+  // which is used by MPS backend.
+  id mpsCD = NSClassFromString(@"MPSGraph");
+  if ([mpsCD instancesRespondToSelector:@selector(LSTMWithSourceTensor:
+                                                           recurrentWeight:
+                                                               inputWeight:
+                                                                      bias:
+                                                                 initState:
+                                                                  initCell:
+                                                                descriptor:
+                                                           name:)] == NO)) {
```
Comment (malfet) [inline aten/src/ATen/mps/MPSDevice.mm:35]: This should fix the build
```suggestion
                                                           name:)] == NO) {
```

**Review by malfet — CHANGES_REQUESTED** [review]
But let's not land this one, but rather fix error in the original PR

**Comment by kulinseth** [comment]
> @kulinseth can you please elaborate, what fix were applied?(I've committed suggested fix into the PR) Also, please note that one must add `ciflow/trunk` label to get signal on the builds

This is my bad . I had used the GitHub ‘s feature of applying the suggested change using their web interface itself in previous pr . That caused build failure . @malfet i thought I had the fix you uploaded in my  commit before . As I locally built and ran all the tests . I will check if I missed it

**Comment by kulinseth** [comment]
https://github.com/pytorch/pytorch/pull/78354, this has the re-land changes.

**Pull request closed without merging**